FLT/GlobalLanglandsConjectures/GLnDefs.lean

```diff
@@ -242,6 +242,13 @@ def _root_.RingHom.GL {A B : Type*} [CommRing A] [CommRing B] (φ : A →+* B)
   (m : Type*) [Fintype m] [DecidableEq m] :
   GL m A →* GL m B := Units.map <| (RingHom.mapMatrix φ).toMonoidHom
 
+structure InvariantSubgroup (U : Subgroup (GL (Fin n) (FiniteAdeleRing ℤ ℚ)))
+  (f : (GL (Fin n) (FiniteAdeleRing ℤ ℚ)) × (GL (Fin n) ℝ) → ℂ) : Prop where
+  is_open : IsOpen U.carrier
+  is_compact : IsCompact U.carrier
+  finite_level (u : U.carrier) (x : GL (Fin n) (FiniteAdeleRing ℤ ℚ)) (y : GL (Fin n) ℝ) :
+    f (u * x, y) = f (x, y)
```

Comment:
I apologise if I've written it wrong somewhere, but this should be f(xu,y) not f(ux,y).